app/actors/ClientSocketActor.scala

```diff
@@ -0,0 +1,62 @@
+package actors
+import akka.actor._
+import actors.PublishSocketMessageActor.{AddOutActor, PublishMessage, StopMessage}
+import models.EventEnums.EventType
+import models.{EventJsonSupport, SocketMessage}
+import spray.json._
+
+import scala.collection.mutable.HashSet
+import play.api.libs.json._
+import play.api.libs.functional.syntax._
+
+object ClientSocketActor {
+  def props(out: ActorRef, publisher: ActorRef): Props = Props(new ClientSocketActor(out, publisher))
+}
+
+class ClientSocketActor(out: ActorRef, publisher: ActorRef) extends Actor with EventJsonSupport {
+
+  val myEvents: HashSet[EventType] = HashSet.empty[EventType]
+
+  implicit val messageReads: Reads[SocketMessage] = ((JsPath \ "event").read[EventType] and
+  (JsPath \ "payload").readNullable[String])(SocketMessage.apply _)
+  implicit val messageWrites: Writes[SocketMessage] = Json.writes[SocketMessage]
+
+  override def preStart() {
+    println("pre start called in client", self)
```

Comment:
Please use logging instead of `println` throughout.

Reply:
switchted from println to logging as requested